Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -1069,22 +1069,30 @@ finitely supported functions from `β` to `M`,
 protected def uncurry (f : α →₀ β →₀ M) : α × β →₀ M :=
   f.sum fun a g => g.sum fun b c => single (a, b) c
 
+@[simp]
+theorem uncurry_apply (f : α →₀ β →₀ M) (x : α) (y : β) : f.uncurry (x, y) = f x y := by
```

Comment:
Note existing `Function.uncurry_apply_pair`. I think you probably want to make this protected and add the `_pair`.